identity-account-storage/src/storage/memstore.rs

```diff
@@ -245,6 +255,90 @@ impl Storage for MemStore {
     }
   }
 
+  async fn encrypt_data(
+    &self,
+    did: &IotaDID,
+    data: Vec<u8>,
+    associated_data: Vec<u8>,
+    encryption_options: &EncryptionOptions,
+    private_key: &KeyLocation,
+    public_key: PublicKey,
+  ) -> Result<EncryptedData> {
+    // Retrieves the PrivateKey from the vault
+    let vaults: RwLockReadGuard<'_, _> = self.vaults.read()?;
+    let vault: &MemVault = vaults.get(did).ok_or(Error::KeyVaultNotFound)?;
+    let key_pair: &KeyPair = vault.get(private_key).ok_or(Error::KeyNotFound)?;
+    // Encrypts the data
+    match key_pair.type_() {
+      KeyType::Ed25519 => Err(Error::InvalidPrivateKey(
+        "ED25519 keys are not suported for encrypting/decrypting data".to_owned(),
```

Comment:
@PhilippGackstatter flagged these error messages elsewhere so I'll highlight these for consistency:

```suggestion
        "Ed25519 keys are not supported for encryption".to_owned(),
```